src/Serialization/main.jl

```diff
@@ -653,13 +653,12 @@ function load(io::IO; params::Any = nothing, type::Any = nothing,
   end
 
   if file_version < VERSION_NUMBER
-    jsondict = JSON.parse(json(s.obj), dicttype=Dict{Symbol, Any})
+    jsondict = JSON.parse(JSON3.write(s.obj), dicttype=Dict{Symbol, Any})
```

Comment:
```suggestion
    # we need a mutable dictionary
    jsondict = copy(s.obj)
```
I would prefer to do it this way but that might not work as expected if `s.obj` is not from `JSON3`. Maybe we need some helper method to extract the json as proper dict which dispatches on the different `s.obj` types ?